File: Examples/DeveloperObjcExample/ObjcExample/GroupViewController.m
//
//  GroupViewController.m
//  AppcuesObjcExample
//
//  Created by Matt on 2022-06-01.
//

#import "GroupViewController.h"
#import "Appcues+Shared.h"

@interface GroupViewController ()
@property (weak, nonatomic) IBOutlet UITextField *groupIDTextField;

@end

@implementation GroupViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    [[Appcues shared] screenWithTitle:@"Update Group" properties:nil];
}

- (IBAction)saveGroupTapped:(UIButton *)sender {
    [self.view endEditing:YES];
    [[Appcues shared] groupWithGroupID: _groupIDTextField.text properties:@{@"test_user":@YES}];
}

@end

File: Examples/DeveloperCocoapodsExample/CocoapodsExample/GroupViewController.swift
//
//  GroupViewController.swift
//  AppcuesCocoapodsExample
//
//  Created by James Ellis on 12/14/21.
//

import UIKit
import AppcuesKit

class GroupViewController: UIViewController {
    @IBOutlet private var groupIDTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        Appcues.shared.screen(title: "Update Group")
    }

    @IBAction private func saveGroupTapped(_ sender: Any) {
        view.endEditing(true)
        Appcues.shared.group(groupID: groupIDTextField.text, properties: [
            "test_user": true
        ])
    }
}
